core-backend/common/src/utils.rs

```diff
@@ -87,25 +58,55 @@ fn smart_truncate(s: &mut String, max_bytes: usize) {
     }
 }
 
-#[derive(Debug, Copy, Clone, derive_more::Display)]
-pub struct LimitedStr<'a>(&'a str);
+/// Wrapped string to fit `core-backend::TRIMMED_MAX_LEN` amount of bytes.
+///
+/// The `Cow` is used to avoid allocating a new `String` when the `LimitedStr` is
+/// created from a `&str`.
+///
+/// Plain `str` is not used because it can't be properly encoded/decoded via scale codec.
+#[derive(
+    TypeInfo, Encode, Decode, Debug, Clone, derive_more::Display, PartialEq, Eq, PartialOrd, Ord,
+)]
+#[codec(crate = scale)]
```

Comment:
Could you explain me: why do we need this #[codec(crate = scale)] ?

Reply:
> Could you explain me: why do we need this #[codec(crate = scale)] ?

It allows don't depend from parity_scale_codec, but at the same time use their derive macros from `pub extern crate codec` of other libs.
That caused by internal implementation of these derive macros.

Reply:
But.. as I can see scale is parity scale codec:
https://github.com/paritytech/scale-info/blob/cc97f4c6496acf0f92d9caa45856331f094a7f38/Cargo.toml#L23

So, why we need to specify this, if it's by default